src/hotspot/share/jeandle/jeandleAbstractInterpreter.hpp

```diff
@@ -80,18 +107,20 @@ class JeandleVMState : public JeandleCompilationResourceObj {
   llvm::Value* dpop() { return pop(BasicType::T_DOUBLE); }
 
   // Untyped manipulation (for dup_x1, etc.)
-  void raw_push(llvm::Value* t) { _stack.push_back(t); }
-  llvm::Value* raw_pop() { llvm::Value* v = _stack.back(); _stack.pop_back(); return v; }
+  void raw_push(BasicType bt, llvm::Value* t) { _stack.push_back(TypedValue(bt, t)); }
+  void raw_push(TypedValue tv) { _stack.push_back(tv); }
```

Comment:
I think only the second interface should be reserved, to keep the interfaces clean.